st2actions/st2actions/container/base.py

```diff
@@ -96,18 +96,15 @@ def _do_run(self, runner, runnertype_db, action_db, actionexec_db):
             runner.pre_run()
 
             LOG.debug('Performing run for runner: %s', runner)
-            if isinstance(runner, AsyncActionRunner):
-                (status, result, query_context) = runner.run(action_params)
-            else:
-                (status, result) = runner.run(action_params)
+            (status, result, context) = runner.run(action_params)
 
             try:
                 result = json.loads(result)
             except:
                 pass
 
-            if status not in DONE_STATES:
-                self._setup_async_query(actionexec_db.id, runnertype_db, query_context)
+            if isinstance(runner, AsyncActionRunner) and status not in DONE_STATES:
```

Comment:
These are redundant checks 

- Why would any runner which is not Async return a non- DONE state?
- What happens if a non Async runner returns a non- DONE state? Do we really care if it was a non Async runner in the first place?